version: '3.8'

services:
  web:
    build: .
    ports:
      - "8000:8000"
    volumes:
      - .:/app
    environment:
      - DB_DRIVER=postgresql
      - DB_USER=postgres
      - DB_PASSWORD=postgres
      - DB_HOST=db  # Используем имя сервиса вместо localhost
      - DB_PORT=5432
      - DB_NAME=fastapi_db
      - SECRET_KEY=your-secret-key
      - PYTHONUNBUFFERED=1
    command: uvicorn app.api.main:app --host 0.0.0.0 --port 8000 --reload
    depends_on:
      - db
    restart: unless-stopped
    networks:
      - backend-network

  db:
    image: postgres:13
    environment:
      - POSTGRES_USER=postgres  # Должен совпадать с DB_USER
      - POSTGRES_PASSWORD=postgres  # Должен совпадать с DB_PASSWORD
      - POSTGRES_DB=fastapi_db  # Должен совпадать с DB_NAME
    volumes:
      - postgres_data:/var/lib/postgresql/data
    ports:
      - "5432:5432"
    restart: unless-stopped
    networks:
      - backend-network
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 5s
      timeout: 5s
      retries: 5

volumes:
  postgres_data:

networks:
  backend-network:
    driver: bridge